### PR TITLE
Feature/api calls

### LINKED ISSUE #13: As a user, I will see proper error handling
As a user, I should see an error message if the server returns a 500 response, so that I know the issue is with the server 

Given the event that the server returns a 500 response, the error message will be clearly displayed on the DOM for the user to see and easily understand

### LINKED ISSUE #14: Fetch Data from API
Refactor App to fetch data from API, rather than using data file provided

### LINKED ISSUE #16: First Weekend Deliverables
1. Iterations 0-3 completed
2. Update project board so functionality is described as user stories
3. Each user story includes thorough acceptance criteria
4. Tag instructors in one PR (refer to spec, should be a feature being refactored or tests being written)

### PR DESCRIPTION
**Pull request by michannstoner**
##### Description
What's changed & where should the reviewer start? 
Brought in fetched data instead of using local data
- apiCalls.js
- App.js
- MovieDetails.js

#### Type of change
What is it?
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring
### Checklist:
- [x] My code follows the guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] Any dependent changes have been merged and published in downstream module

### CODE REVIEW

**Review by sarahlane8 — APPROVED** [review]
This looks great! I'd be interested to research async and await and see how that would be implemented instead of .then.

I like the error message rendering. We will have to ensure the user has a way to refresh or go back to homepage, etc, if an error occurs before the final product is completed.

File: src/apiCalls.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react'
```
Comment (letakeane) [inline src/apiCalls.js:1]: I don't think you need this line here 😄

File: src/components/App/App.js [hunk 2]
```diff
@@ -1,38 +1,54 @@
...
+// import movieData from '../../movieData'
 import './App.css'
+import { getAllMovies, getSingleMovie } from '../../apiCalls'
 
 class App extends Component {
   constructor() {
     super()
     this.state = {
-      movieData: movieData.movies,
+      movieData: [],
```
Comment (letakeane) [inline src/components/App/App.js:12]: Nice starting value here

File: src/components/App/App.js [hunk 3]
```diff
@@ -1,38 +1,54 @@
...
     }
   }
 
-  toggleMovieDetails = id => {
+  componentDidMount = () => {
+    getAllMovies()
+      .then(data => {
+        this.setState({ movieData: data.movies })
+      })
+      .catch(error => this.setState({ error: 'Something went wrong, try again later!' }))
```
Comment (letakeane) [inline src/components/App/App.js:24]: Good job setting this error in state

File: src/components/App/App.js [hunk 4]
```diff
@@ -1,38 +1,54 @@
...
+  }
 
-    this.setState({displayMovieDetails: !this.state.displayMovieDetails, id: id})
+  toggleMovieDetails = id => {
```
Comment (letakeane) [inline src/components/App/App.js:27]: I wonder if this function is necessary. Seems like it simply calls `fetchSingleMovie` - perhaps whatever invokes this function could simply invoke `fetchSingleMovie` instead.

File: src/components/App/App.js [hunk 5]
```diff
@@ -1,38 +1,54 @@
...
+      })
+      .catch(error => this.setState({ error: 'Something went wrong, try again later!' }))
   }
 
   render() {
     return (
       <main className="App">
         <h1>Rancid Tomatillos</h1>
-        {!this.state.displayMovieDetails &&
+        {this.state.error && <h2>{this.state.error}</h2>}
```
Comment (letakeane) [inline src/components/App/App.js:43]: Nice conditional rendering.

**Review by letakeane — COMMENTED** [review]
Nice work! It's clear you're building a clear understanding of React 😄

**Pull request merged**